tests/unit_tests/commandline/cli_top_level_tester.py

```diff
@@ -125,33 +262,139 @@ def check_output(self, args, output):
         """
         pass
 
-    def test_all_command_combinations(self, args=None):
-        """Test all 'geoips list <sub-cmd> ...' commands.
+    def capture_output(self, func, args):
+        """Redirect stdout and stderr output from a python function to two variables.
+
+        Where these two variables (stdout, stderr), will be used for testing the output
+        of CLI commands to make sure they're working as expected.
+
+        Parameters
+        ----------
+        func: python function
+            - The function whose output we want to capture.
+        args: array of str
+            - List of arguments to call the CLI with (ie. ['geoips', '<cmd>'])
+
+        Returns
+        -------
+        stdout: str
+            - The ouput that would have been printed to sys.stdout.
+        stderr: str
+            - The output that would have been printed to sys.stderr.
+        """
+        stdout_capture = io.StringIO()
+        stderr_capture = io.StringIO()
+
+        with (
+            contextlib.redirect_stdout(stdout_capture),
+            contextlib.redirect_stderr(stderr_capture),
+        ):
+            try:
+                func()
+            except SystemExit as se:
+                print(se, file=sys.stderr)
+            except Exception as e:
+                print(e, file=sys.stderr)
+            # except RuntimeError as re:
+            #     print(re, file=sys.stderr)
+
+        # Retrieve the output printed from 'func'
+        stdout = stdout_capture.getvalue()
+        stderr = stderr_capture.getvalue()
+
+        # Close the StringIO Objects
+        stdout_capture.close()
+        stderr_capture.close()
```

Comment:
This looks like it should work, but is it duplicating `pytest.capsys` and other similar fixtures from pytest? https://docs.pytest.org/en/7.1.x/how-to/capture-stdout-stderr.html

No need to change this now but, if this does duplicate capabilities from pytest, create an issue to switch to the pytest fixtures.

Reply:
I'll play around with this and see if ``pytest.capsys`` works in another PR. I'll create an issue for this as well.

Reply:
See #628.